# Description: Add Oxide classes to base::ThreadRestrictions friends so that
#  they can use ScopedAllowWait
# Author: Chris Coulson <chris.coulson@canonical.com>

diff --git a/base/threading/thread_restrictions.h b/base/threading/thread_restrictions.h
--- a/base/threading/thread_restrictions.h
+++ b/base/threading/thread_restrictions.h
@@ -58,16 +58,20 @@ class WatcherThreadManager;
 }
 namespace net {
 class NetworkChangeNotifierMac;
 namespace internal {
 class AddressTrackerLinux;
 }
 }
 
+namespace oxide {
+class CompositorUtilsImpl;
+}
+
 namespace remoting {
 class AutoThread;
 }
 
 namespace ui {
 class WindowResizeHelperMac;
 }
 
@@ -190,16 +194,18 @@ class BASE_EXPORT ThreadRestrictions {
   friend class MessagePumpDefault;
   friend class SequencedWorkerPool;
   friend class SimpleThread;
   friend class Thread;
   friend class ThreadTestHelper;
   friend class PlatformThread;
   friend class android::JavaHandlerThread;
 
+  friend class oxide::CompositorUtilsImpl;
+
   // END ALLOWED USAGE.
   // BEGIN USAGE THAT NEEDS TO BE FIXED.
   friend class ::chromeos::BlockingMethodCaller;  // http://crbug.com/125360
   friend class ::chromeos::system::StatisticsProviderImpl;  // http://crbug.com/125385
   friend class chrome_browser_net::Predictor;     // http://crbug.com/78451
   friend class
       content::BrowserGpuChannelHostFactory;      // http://crbug.com/125248
   friend class
